main.go

```diff
@@ -60,6 +60,14 @@ func main() {
 	}
 	v = append(v, fmt.Sprintf("spec: %s", specs.Version))
 	app.Version = strings.Join(v, "\n")
+
+	root := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
This is not correct, as it will mean that if you run `runc` as root with `XDG_RUNTIME_DIR` set, then the spawned containers won't be visible by other `runc` management processes. Instead, this should be conditional based on whether you're an unprivileged user. Even then I'm still not convinced, because what if someone sets up `/run/runc` to be like `/tmp` (sticky-bit and world-writeable) in order for all users to access the same store of container state?

I agree that `runc` should obey `XDG_RUNTIME_DIR`, but I'm not entirely convinced that this is correct (especially since changing `XDG_RUNTIME_DIR` will make it more difficult for other people to manage the `runc` containers spawned by someone with `XDG_RUNTIME_DIR` set.

@crosbymichael do you have any ideas on how this should work nicely?